### PR TITLE
internal/pkg/agent/cmd: don't format error message with nil errors

### PR DESCRIPTION
**Pull request by efd6**
## What does this PR do?

This fixes formatting of errors resulting from non-error states but non-accepted HTTP status codes.

The failure conditions allow nil errors to result in an error being formatted, when formatting due to a non-accepted HTTP status code and a nil error, omit the error.

## Why is it important?

Reduces log syntax noise.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

File: internal/pkg/agent/cmd/container.go [hunk 1]
```diff
@@ -653,8 +653,13 @@ func performGET(cfg setupConfig, client *kibana.Client, path string, response in
...
 		code, result, err := client.Connection.Request("GET", path, nil, nil, nil)
 		if err != nil || code != 200 {
-			err = fmt.Errorf("http GET request to %s%s fails: %w. Response: %s",
-				client.Connection.URL, path, err, truncateString(result))
+			if err != nil {
+				err = fmt.Errorf("http GET request to %s%s fails: %w. Response: %s",
+					client.Connection.URL, path, err, truncateString(result))
+			} else {
+				err = fmt.Errorf("http GET request to %s%s fails. Response: %s",
+					client.Connection.URL, path, truncateString(result))
```
Comment (cmacknz) [inline internal/pkg/agent/cmd/container.go:661]:
```suggestion
				err = fmt.Errorf("http GET request to %s%s fails. StatusCode: %d Response: %s",
					client.Connection.URL, path, code, truncateString(result))
```

File: internal/pkg/agent/cmd/container.go [hunk 2]
```diff
@@ -672,8 +677,13 @@ func performPOST(cfg setupConfig, client *kibana.Client, path string, writer io.
...
 		code, result, err := client.Connection.Request("POST", path, nil, nil, nil)
 		if err != nil || code >= 400 {
-			err = fmt.Errorf("http POST request to %s%s fails: %w. Response: %s",
-				client.Connection.URL, path, err, truncateString(result))
+			if err != nil {
+				err = fmt.Errorf("http POST request to %s%s fails: %w. Response: %s",
+					client.Connection.URL, path, err, truncateString(result))
+			} else {
+				err = fmt.Errorf("http POST request to %s%s fails. Response: %s",
+					client.Connection.URL, path, truncateString(result))
```
Comment (cmacknz) [inline internal/pkg/agent/cmd/container.go:685]:
```suggestion
				err = fmt.Errorf("http POST request to %s%s fails. StatusCode: %d Response: %s",
					client.Connection.URL, path, code, truncateString(result))
```

**Review by cmacknz — APPROVED** [review]
Let's log the HTTP status code in the non-error case, otherwise LGTM.

**Comment by efd6** [comment]
The status code is already included in the response, e.g.
```
Kibana Fleet setup failed: http POST request to https://kibana:5601/api/fleet/setup fails: [Elastic-Agent (elastic-package)] could not be added. [system] could not be installed due to error: [PackageNotFoundError: [system] package not found in registry]: %!w(<nil>). Response: {"statusCode":500,"error":"Internal Server Error","message":"[Elastic-Agent (elastic-package)] could not be added. [system] could not be installed due to error: [PackageNotFoundError: [system] package not found in registry]"}
```

File: internal/pkg/agent/cmd/container.go (hunk 1 above)
Comment (AndersonQ) [inline internal/pkg/agent/cmd/container.go:661]: [Suggestion]
I'd rather not rely on the server returning the status code on the error response, it's more reliable using the one we received.

File: internal/pkg/agent/cmd/container.go (hunk 2 above)
Comment (AndersonQ) [inline internal/pkg/agent/cmd/container.go:685]: [Suggestion]
I'd rather not rely on the server returning the status code on the error response, it's more reliable using the one we received.

**Review by AndersonQ — APPROVED** [review]

File: internal/pkg/agent/cmd/container.go (hunk 1 above)
Comment (efd6) [reply internal/pkg/agent/cmd/container.go:661]: OK

**Pull request merged**